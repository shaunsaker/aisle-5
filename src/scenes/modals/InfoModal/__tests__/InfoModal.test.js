import React from 'react';
import renderer from 'react-test-renderer';

import { InfoModal } from '..';

// Fixes _bezier is not a function bug
jest.useFakeTimers();

describe('InfoModal', () => {
  const spies = [];
  const dispatch = jest.fn();

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<InfoModal />);

      expect(component).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('should handle onClose', () => {
      spies[0] = jest.spyOn(InfoModal.prototype, 'navigate');
      const component = renderer.create(<InfoModal />);
      const instance = component.getInstance();

      instance.onClose();

      expect(spies[0]).toHaveBeenCalled();
    });

    it('should handle navigate', () => {
      const component = renderer.create(<InfoModal />);
      const instance = component.getInstance();

      instance.navigate();
    });
  });

  describe('actions', () => {
    spies[0] = jest.spyOn(InfoModal.prototype, 'onClose');
    const component = renderer.create(<InfoModal />);
    const { root } = component;
    const targetComponent = root.findByProps({ testID: 'infoModal.button.submit' });

    targetComponent.props.handlePress();

    expect(spies[0]).toHaveBeenCalled();
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
    dispatch.mockClear();
  });
});
