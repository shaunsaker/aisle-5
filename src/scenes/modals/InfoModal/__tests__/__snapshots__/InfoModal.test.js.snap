// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoModal renders renders with minimum required props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.67)",
      "bottom": 0,
      "left": 0,
      "opacity": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "flex": 1,
          },
          Object {},
        ]
      }
    >
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        bounces={false}
        contentContainerStyle={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutomaticScroll={true}
        enableOnAndroid={false}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="interactive"
        keyboardOpeningTime={250}
        keyboardShouldPersistTaps="handled"
        keyboardSpace={0}
        onScroll={[Function]}
        resetKeyboardSpace={[Function]}
        scrollEnabled={false}
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollIntoView={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        update={[Function]}
        viewIsInsideTabBar={false}
      >
        <View>
          <View
            style={
              Object {
                "backgroundColor": "#EDE8CA",
                "borderColor": null,
                "borderRadius": 5,
                "borderWidth": 0,
                "elevation": 12,
                "marginHorizontal": 15,
                "marginVertical": 10,
                "maxWidth": 480,
                "paddingHorizontal": 30,
                "paddingVertical": 20,
                "shadowColor": "#212121",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.33,
                "shadowRadius": 4,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#212121",
                  "fontFamily": "ArchivoBlack-Regular",
                  "fontSize": 24,
                  "lineHeight": 30,
                  "marginBottom": 10,
                  "textAlign": "center",
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#212121",
                  "fontFamily": "Muli-SemiBold",
                  "fontSize": 17,
                  "lineHeight": 25.5,
                  "marginBottom": 20,
                  "textAlign": "center",
                }
              }
            />
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#D17623",
                    "borderColor": null,
                    "borderRadius": 5,
                    "borderWidth": 0,
                    "elevation": 6,
                    "opacity": 1,
                    "paddingHorizontal": 15,
                    "paddingVertical": 10,
                    "shadowColor": "#212121",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.33,
                    "shadowRadius": 2,
                  }
                }
                testID="infoModal.button.submit"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#212121",
                        "fontFamily": "Muli-Bold",
                        "fontSize": 17,
                      },
                      Object {
                        "color": "#EDE8CA",
                      },
                      undefined,
                    ]
                  }
                >
                  Nice
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
